src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -165,21 +267,36 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 }
                 telemetryEventDetails.SetProperty(TelemetryConstants.AssertionType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.AssertionType);
                 telemetryEventDetails.SetProperty(TelemetryConstants.Endpoint, AuthenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority.ToString());
+
                 telemetryEventDetails.SetProperty(TelemetryConstants.CacheLevel, (int)authenticationResult.AuthenticationResultMetadata.CacheLevel);
-                ParseScopesForTelemetry(telemetryEventDetails);
+              
+                Tuple<string, string> resourceAndScopes = ParseScopesForTelemetry();
+                if (resourceAndScopes.Item1 != null)
+                {
+                    telemetryEventDetails.SetProperty(TelemetryConstants.Resource, resourceAndScopes.Item1);
+                }
+
+                if (resourceAndScopes.Item2 != null)
+                {
+                    telemetryEventDetails.SetProperty(TelemetryConstants.Scopes, resourceAndScopes.Item2);
+                }
+
+                
             }
         }
 
-        private void ParseScopesForTelemetry(MsalTelemetryEventDetails telemetryEventDetails)
+        private Tuple<string, string> ParseScopesForTelemetry()
         {
+            string resource = null;
+            string scopes = null;
             if (AuthenticationRequestParameters.Scope.Count > 0)
             {
                 string firstScope = AuthenticationRequestParameters.Scope.First();
```

Comment:
can scopes ever be null? like in a failure case? do we then add metrics for it

Reply:
When the scopes are null, there is a MSAL failure with client exception. So this will not be logged. This is logged for success scenarios.